torch/_functorch/autograd_function.py

```diff
@@ -240,11 +252,165 @@ def custom_function_call_vmap(interpreter, autograd_function, *operands):
     return output
 
 
+def custom_function_call_vmap_generate_rule(interpreter, autograd_function, *operands):
+    unwrapped_operands, in_dims = unwrap_batched(operands, interpreter.level())
+    vmapped_function, get_out_dims = vmapify_autograd_function(
+        autograd_function, in_dims, interpreter.batch_size(), interpreter.randomness())
+
+    with interpreter.lower():
+        output = custom_function_call(vmapped_function, *unwrapped_operands)
+
+    out_dims = get_out_dims()
+    return wrap_batched(output, out_dims, interpreter.level())
+
+
 @custom_function_call.py_impl(TransformType.Functionalize)
-def custom_function_call_functionalize(interpreter, autograd_function, *operands):
+def custom_function_call_functionalize(interpreter, autograd_function, generate_vmap_rule, *operands):
     raise RuntimeError("NYI: Functionalize rule for custom_function_call")
 
 
+def vmapify_autograd_function(autograd_function, in_dims, batch_size, randomness):
+    # The following values are saved from the forward() and setup_context()
+    # and used in backward().
+    # Why do we save the values out here instead of on the ctx object?
+    # - out_dims: There's no way to retrieve this from forward()
+    # - input_shapes, saved_tensors_bdims: I'm a bit scared of nesting
+    #   vmap(vmap( but not completely sure if it is a problem. If we
+    #   assigned those fields to the ctx object, the worry is that they
+    #   get overwritten.
```

Comment:
For input_shapes and saved_tensor_bdims, isn't it less likely to get overridden when you do `grad(grad(vmap` if you save them on ctx instead of here because you make a new ctx for every layer of grad you have.

Currently I would think that it is getting overridden because the two grads refer to the same vmapified autograd_function,  but it just doesn't matter because both grad layers should be saving the same input_shapes + saved-tensor_bdims (because there's no vmap in between). And even if you have something like `grad(vmap(grad(vmap`, that doesn't matter because now the two grads should be referring to different autograd_function objects.

Reply:
If you do grad(grad(vmap, then yeah, those are not being overridden because each grad layer gets a different ctx object.

The concern is for two consecutive vmap, like grad(vmap(vmap. I haven't actually tested this so it *might* just work out of the box, but another benefit of keeping these attributes out here is that I don't need to denylist more attrs for the user. Though there are other ways around the denylist (we can shove all extra attrs we want into a single object on the ctx)